core/src/net/sf/openrocket/aerodynamics/BarrowmanCalculator.java

```diff
@@ -392,84 +392,90 @@ private double calculateFrictionDrag(FlightConfiguration configuration, FlightCo
 		
 		double[] roughnessLimited = new double[Finish.values().length];
 		Arrays.fill(roughnessLimited, Double.NaN);
-		
-		for (RocketComponent c : configuration.getActiveComponents()) {
+
+		final InstanceMap imap = configuration.getActiveInstances();
+	    for(Map.Entry<RocketComponent, ArrayList<InstanceContext>> entry: imap.entrySet() ) {
+			final RocketComponent c = entry.getKey();
 			
 			// Consider only SymmetricComponents and FinSets:
 			if (!(c instanceof SymmetricComponent) &&
 					!(c instanceof FinSet))
 				continue;
+
+				// iterate across component instances
+			final ArrayList<InstanceContext> contextList = entry.getValue();
+			for(InstanceContext context: contextList ) {
 			
-			// Calculate the roughness-limited friction coefficient
-			Finish finish = ((ExternalComponent) c).getFinish();
-			if (Double.isNaN(roughnessLimited[finish.ordinal()])) {
-				roughnessLimited[finish.ordinal()] =
+				// Calculate the roughness-limited friction coefficient
+				Finish finish = ((ExternalComponent) c).getFinish();
+				if (Double.isNaN(roughnessLimited[finish.ordinal()])) {
+					roughnessLimited[finish.ordinal()] =
 						0.032 * Math.pow(finish.getRoughnessSize() / configuration.getLength(), 0.2) *
-								roughnessCorrection;
-			}
+						roughnessCorrection;
+				}
 			
-			/*
-			 * Actual Cf is maximum of Cf and the roughness-limited value.
-			 * For perfect finish require additionally that Re > 1e6
-			 */
-			double componentCf;
-			if (configuration.getRocket().isPerfectFinish()) {
-				
-				// For perfect finish require Re > 1e6
-				if ((Re > 1.0e6) && (roughnessLimited[finish.ordinal()] > Cf)) {
-					componentCf = roughnessLimited[finish.ordinal()];
+				/*
+				 * Actual Cf is maximum of Cf and the roughness-limited value.
+				 * For perfect finish require additionally that Re > 1e6
+				 */
+				double componentCf;
+				if (configuration.getRocket().isPerfectFinish()) {
+					
+					// For perfect finish require Re > 1e6
+					if ((Re > 1.0e6) && (roughnessLimited[finish.ordinal()] > Cf)) {
+						componentCf = roughnessLimited[finish.ordinal()];
+					} else {
+						componentCf = Cf;
+					}
+					
 				} else {
-					componentCf = Cf;
+					
+					// For fully turbulent use simple max
+					componentCf = Math.max(Cf, roughnessLimited[finish.ordinal()]);
+					
 				}
-				
-			} else {
-				
-				// For fully turbulent use simple max
-				componentCf = Math.max(Cf, roughnessLimited[finish.ordinal()]);
-				
-			}
 			
-			//Handle Overriden CD for Whole Rocket
-			if(c.isCDOverridden()) {
-				continue;
-			}
+				//Handle Overriden CD for Whole Rocket
+				if(c.isCDOverridden()) {
```

Comment:
I think this condition could be moved prior to processing the active instances. None of the iterations above this check within the loop contribute to the final calculations returned at the end. Moving this check up will save some unnecessary loop iterations.

Reply:
Another one I'd prefer to fix later

Reply:
+1